param/_utils.py

```diff
@@ -612,3 +612,15 @@ def async_executor(func):
         task.add_done_callback(_running_tasks.discard)
     else:
         event_loop.run_until_complete(func())
+
+
+def anyinstance(obj, class_tuple_generator):
```

Comment:
Could an approach like this be used to avoid creating these functions? https://peps.python.org/pep-3119/#overloading-isinstance-and-issubclass

Reply:
It seems it'd be something like:

```python

import sys

class GeneratorIsMeta(type):
    def __instancecheck__(cls, inst):
        return isinstance(inst, tuple(cls.gen_types()))

    def __subclasscheck__(cls, sub):
        return issubclass(sub, tuple(cls.gen_types()))


class DtTypes(metaclass=GeneratorIsMeta):
    @classmethod
    def gen_types(cls):
        yield dt.datetime
        yield dt.date
        if "numpy" in sys.modules:
            import numpy as np
            yield np.datetime64

class IntTypes(metaclass=GeneratorIsMeta):
    @classmethod
    def gen_types(cls):
        yield int
        if "numpy" in sys.modules:
            import numpy as np
            yield np.integer
```

Reply:
Thank you. 

I still thinks it is pretty advanced (even though you don't need it to be iterable in your example). 

You could likely move the advanced logic into a decorator, but then you would need to import that into our other libraries. My main point with accepting iterator is not so much for param itself but for HoloViews.

Reply:
> My main point with accepting iterator is not so much for param itself but for HoloViews.

Oh yeah I agree users shouldn't have to pass this custom class, a generator seems appropriate. Instead I was wondering if there couldn't be a way to use the metaclass approach internally to avoid the special `anyisinstance` and `anyissubclass` functions, they seem to be easy to forget in the future.

Reply:
I have switched to a metaclass approach with a decorator for easy access.